elfloader-tool/src/plat/tk1/monitor.S

```diff
@@ -64,10 +64,10 @@ smc_handler:
     ldr sp, [pc, #-12]
     push {lr}
     blx  r0
-    mrc p15, 0, r0, c1, c1, 0
+    mrc p15, 0, r3, c1, c1, 0
```

Comment:
If we stick to the calling convention, r3 is also a return result register. Maybe we could use r12 instead?

Reply:
That should work.  I'll send an updated commit.